Sources/ClientRuntime/Networking/Http/HttpResponse.swift

```diff
@@ -4,12 +4,24 @@
  */
 import AwsCommonRuntimeKit
 
-public class HttpResponse: HttpUrlResponse {
+public actor HttpResponse: HttpUrlResponse {
 
     public var headers: Headers
     public var body: ByteStream
     public var statusCode: HttpStatusCode
 
+    public func addHeaders(additionalHeaders: Headers) {
+        self.headers.addAll(headers: additionalHeaders)
+    }
+
+    public func setBody(newBody: ByteStream) {
```

Comment:
Does this function not need to be marked `async`?  We call it with `async`...

Reply:
Unless I'm mistaken, I don't think the functions in an actor have to be marked async unless they call async functions themselves. In this case the setter functions don't call anything async themselves (they just set the properties of the actor instance), so the async keyword can be left out. Though when we call these methods outside of the actor, we need to prepend `await` keyword since the action gets put into actor's queue of tasks that gets completed in async manner.